contracts/Exchange.sol

```diff
@@ -170,6 +170,8 @@ contract Exchange is Restricted {
     function _fillOrder(uint64 buyTokenId, uint64 sellTokenId) private {
         Order storage buy = buyTokenOrders[buyTokenId];
         Order storage sell = sellTokenOrders[sellTokenId];
+        require(buy.amount > 0, "buy order must have unfilled amount left");
```

Comment:
Should the matching process skip canceled / fulfilled orders silently, instead of reverting? That seems to preserve the intention better on a best-effort basis.

Reply:
good call. that way matchmultiple could carry on iterating too.
shall we return a true/false result from fillorder fx?

Reply:
Yes, that could work. `_fillOrder` would just return a bool, multi-match skips, single-match reverts.